### PR TITLE
CSHARP-5423: Remove deprecated field from GridFS unified tests

### PR DESCRIPTION
**Pull request by papafe**

### CODE REVIEW

**Comment by papafe** [comment]
Not sure if we want to keep them, but for now I've moved the legacy tests inside the "legacy" folder.

**Review by rstam — CHANGES_REQUESTED** [review]
Please make the commit tree linear before we review.

**Comment by papafe** [comment]
@rstam Rebased on main

**Comment by papafe** [comment]
Converted it to draft as I need to finish a couple of things

**Comment by papafe** [comment]
@BorisDog regarding the deprecated fields, namely "contentType" and "aliases", I've removed them from the registered members of the serializers and also from a test method input given they were not used. 
Apart from that, no need to do more, since all the extra members are just kept into `GridFSFileInfo.BackingDocument` that is a `BsonDocument` and don't cause any issues in case they are encountered. This test verifies that: https://github.com/mongodb/mongo-csharp-driver/blob/476af2b06a102adef9b1ef12c567562069518e79/tests/MongoDB.Driver.Tests/GridFS/GridFSFileInfoTests.cs#L74

**Review by BorisDog — CHANGES_REQUESTED** [review]
Looks good overall.
There was some confusion with the legacy tests, as I thought the spec contains both `legacy` and `unified` folders.
We need to align with the spec tests structure, and have just unified tests (and remove legacy runners).
So we need to do [CSHARP-3587](https://jira.mongodb.org/browse/CSHARP-3587) as prerequisite. We could also address CSHARP-3587 as part of this ticket.

File: tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedGridFsDownloadByNameOperation.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+﻿/* Copyright 2020-present MongoDB Inc.
```
[the remark this answers is not in the document]
Comment (papafe) [reply tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedGridFsDownloadByNameOperation.cs]: Corrected. I noticed that there are other files with various years, mostly in the test project.

**Comment by papafe** [comment]
@BorisDog, I've removed the legacy tests and their runners, so I suppose [CSHARP-3587](https://jira.mongodb.org/browse/CSHARP-3587) could also be closed when this is merged

**Review by BorisDog — APPROVED** [review]
LGTM!

**Pull request merged**